### PR TITLE
Koios benchmarks (2nd PR)

### PR DESCRIPTION
**Pull request by aman26kbm**
#### Description
This is the second PR for Koios benchmarks. The main changes in this PR are:
1. Moving complex_dsp macro out from benchmark verilog files into a separate include file.
2. Updated regression task files to include this new include file.
3. Added new tests that run these benchmarks without the complex_dsp macro defined.
4. Updated documentation to cleanly describe the usage of these benchmarks (including measuring QoR)

#### Related Issue
This PR contains changes pending from the first PR: https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/1753.
This PR also contains changes suggested in the issue: https://github.com/verilog-to-routing/vtr-verilog-to-routing/issues/1782. 

#### Types of changes

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by aman26kbm** [comment]
Tagging @vaughnbetz to please review the changes.

File: vtr_flow/tasks/regression_tests/vtr_reg_weekly/koios/config/config.txt
Comment (vaughnbetz) [inline vtr_flow/tasks/regression_tests/vtr_reg_weekly/koios/config/config.txt:28]: Good to comment that this will use the advanced dsp block mode

File: vtr_flow/tasks/regression_tests/vtr_reg_weekly/koios_no_complex_dsp/config/config.txt
Comment (vaughnbetz) [inline vtr_flow/tasks/regression_tests/vtr_reg_weekly/koios_no_complex_dsp/config/config.txt:30]: Again, doesn't this turn on the advanced dsp block?  Seems to conflict with the name of the task.

**Review by vaughnbetz — CHANGES_REQUESTED** [review]
Some requests for comments and clarification on what is an advanced dsp block vs. not an advanced dsp block test.

File: ODIN_II/regression_test/benchmark/task/koios/task.conf
Comment (aman26kbm) [reply ODIN_II/regression_test/benchmark/task/koios/task.conf:15]: Right. Fixing this now.

File: vtr_flow/tasks/regression_tests/vtr_reg_nightly_test4/koios_multi_arch/config/config.txt
Comment (aman26kbm) [reply vtr_flow/tasks/regression_tests/vtr_reg_nightly_test4/koios_multi_arch/config/config.txt:31]: Adding the comment

File: vtr_flow/tasks/regression_tests/vtr_reg_nightly_test4/koios_no_complex_dsp/config/config.txt
Comment (aman26kbm) [reply vtr_flow/tasks/regression_tests/vtr_reg_nightly_test4/koios_no_complex_dsp/config/config.txt:12]: Adding the comment

File: vtr_flow/tasks/regression_tests/vtr_reg_strong/koios/config/config.txt
Comment (aman26kbm) [reply vtr_flow/tasks/regression_tests/vtr_reg_strong/koios/config/config.txt:22]: Adding the comment

File: vtr_flow/tasks/regression_tests/vtr_reg_strong/koios_no_complex_dsp/config/config.txt
Comment (aman26kbm) [reply vtr_flow/tasks/regression_tests/vtr_reg_strong/koios_no_complex_dsp/config/config.txt:22]: Oops. This is a copy-paste issue. I will need to regenerate the golden results too.

File: vtr_flow/tasks/regression_tests/vtr_reg_weekly/koios/config/config.txt
Comment (aman26kbm) [reply vtr_flow/tasks/regression_tests/vtr_reg_weekly/koios/config/config.txt:28]: Adding the comment

File: vtr_flow/tasks/regression_tests/vtr_reg_weekly/koios_no_complex_dsp/config/config.txt
Comment (aman26kbm) [reply vtr_flow/tasks/regression_tests/vtr_reg_weekly/koios_no_complex_dsp/config/config.txt:30]: Right. Fixing this.

**Comment by aman26kbm** [comment]
Thanks for the review, @vaughnbetz . 
I've pushed a commit that fixes those issues.
I will push another PR in a couple days that will have the updated golden results.

**Comment by vaughnbetz** [comment]
OK, waiting for the new golden results and then will merge.

**Comment by aman26kbm** [comment]
Pushed a commit with the golden results of no_complex_dsp weekly regression. 

No other commit is pending now. So, the changes can be merged after all the checks pass.

**Pull request merged**
